src/Umbraco.Web/Mvc/UmbracoControllerFactory.cs

```diff
@@ -13,12 +16,27 @@ namespace Umbraco.Web.Mvc
     public abstract class UmbracoControllerFactory : IFilteredControllerFactory
     {
         private readonly OverridenDefaultControllerFactory _innerFactory = new OverridenDefaultControllerFactory();
+        private readonly IEnumerable<IRenderController> _renderControllers;
+
+        public UmbracoControllerFactory(IEnumerable<IRenderController> renderControllers)
+        {
+            _renderControllers = renderControllers;
+        }
 
         public abstract bool CanHandle(RequestContext request);
 
         public virtual Type GetControllerType(RequestContext requestContext, string controllerName)
         {
-            return _innerFactory.GetControllerType(requestContext, controllerName);
+            var controllerType = _innerFactory.GetControllerType(requestContext, controllerName);
+            if (controllerType == null)
+            {
+                controllerType = _renderControllers
```

Comment:
I'm unsure about the addition of this code, is it necessary? We didn't require this code before for route hijacking so not sure why we require it now.

Reply:
It's needed because the controllers are only globally locatable via their actual names, but we need a way to say "if one isn't found, also look for controllers with the relevant attribute". From what I could find, the looking up all happens inside MS core code, so the only place I could think to add the additional lookup was here, using controllers in DI. 

This code is locating the relevant controller, then the other code `RenderRouteHandler` digs deeper to look for the relevant action.